### DISCUSSION TITLE
How can I access the current (development/production) mode from within a plugin?

### DISCUSSION THREAD

**Discussion by swissmanu** [webpack/webpack#14585] [Q&A] [answered]
Hey community

I want to get the `mode` option of the current Webpack compilation from within a plugin. Is there a simple way for this without passing it explicitly as option when creating the plugin instance?

Thanks for any hints 🙏

**Comment by vankop** [comment, marked as answer]
you can read property `compiler.options.mode` within plugin

**Reply by swissmanu** [reply]
oh boy, it must be too early in the morning... i just checked the `Compiler` type and could not find anything like this. 🙈

thank you!
